src/server/services/procedures/salio/salio.js

```diff
@@ -0,0 +1,749 @@
+/**
+ * SalIO is meant to be used with the SalIO app for android.
+ * It allows your android device to be used as a sensor which is accessible from inside Netsblox.
+ * 
+ * @alpha
+ * @service
+ */
+
+/*
+ * Based on the RoboScape procedure.
+ *
+ * Sensor to server messages:
+ *  mac_addr[6] time[4] 'I': identification, sent every second
+ *  mac_addr[6] time[4] 'A' x[4] y[4] z[4]: accelerometer data
+ *
+ * Server to sensor messages:
+ *  'A': get accelerometer snapshot
+ *
+ * Environment variables:
+ *  SALIO_PORT: set it to the UDP port (1975) to enable this module
+ *  SALIO_MODE: sets the NetsBlox interface type, can be "security",
+ *      "native" or "both" (default)
+ */
+
+'use strict';
+
+const {PromiseSocket} = require('promise-socket');
+
+const logger = require('../utils/logger')('SalIO');
+const Sensor = require('./sensor');
+const acl = require('../roboscape/accessControl');
+const common = require('./common');
+var dgram = require('dgram'),
+    server = dgram.createSocket('udp4'),
+    SALIO_MODE = process.env.SALIO_MODE || 'both';
+
+const SALIO_TCP_PORT = 8889; // tcp port used by the SalIO app
+
+/*
+ * SalIO - This constructor is called on the first
+ * request to an RPC from a given room.
+ * @constructor
+ * @return {undefined}
+ */
+var SalIO = function () {
+    this._state = {
+        registered: {}
+    };
+};
+
+SalIO.serviceName = 'SalIO';
+// keeps a dictionary of sensor objects keyed by mac_addr
+SalIO.prototype._sensors = {};
+
+SalIO.prototype._ensureLoggedIn = function() {
+    if (this.caller.username !== undefined)
+        throw new Error('Login required.');
+};
+
+SalIO.prototype._ensureAuthorized = async function(sensorId) {
```

Comment:
Not sure if this really buys us much (since it is called once and simply invokes a method w/ the same name and 2 args). Up to you if you want to change it - just a thought.

Reply:
Yeah, I didn't look at the copied stuff's usage. I'll inline that.